hedera-node/src/main/java/com/hedera/services/context/SideEffectsTracker.java

```diff
@@ -274,6 +273,24 @@ public void trackAutoAssociation(final TokenID token, final AccountID account) {
 		autoAssociations.add(association);
 	}
 
+	/**
+	 * Tracks the reward payment to the account.
+	 *
+	 * @param accountNum
+	 * 		Account number of the account receiving the reward.
+	 * @param amount
+	 * 		Reward amount.
+	 */
+	public void trackRewardPayment(final long accountNum, final long amount) {
```

Comment:
We can return immediately if `amount == 0` and then we'll know every value in the `rewardAmounts` is strictly positive.